src/NHibernate.Test/NHSpecificTest/NH3954/ProxyCacheFixture.cs

```diff
@@ -1,34 +1,36 @@
-﻿using System.Collections.Concurrent;
+﻿using System;
+using System.Collections.Concurrent;
 using System.Reflection;
+using System.Runtime.Serialization;
 using NHibernate.Proxy;
 using NHibernate.Proxy.DynamicProxy;
 using NUnit.Framework;
 
 namespace NHibernate.Test.NHSpecificTest.NH3954
 {
-	[TestFixture, Explicit("Demonstrates bug impact on cache, but which tests will fail is a bit unpredictable")]
+	[TestFixture, Explicit("Demonstrates bug impact on cache, but which tests will fail is a bit unpredictable"), Obsolete]
 	public class ProxyCacheFixture
 	{
 		private ProxyCache _cache;
-		private ConcurrentDictionary<ProxyCacheEntry, System.Type> _internalCache;
+		private ConcurrentDictionary<ProxyCacheEntry, TypeInfo> _internalCache;
```

Comment:
The test fixture itself was failing (but because of `[Explicit]` attribute it did not run and we did not notice)